packages/node-http-handler/src/node-http-handler.ts

```diff
@@ -39,18 +39,19 @@ interface ResolvedNodeHttpHandlerConfig {
 
 export class NodeHttpHandler implements HttpHandler {
   private config?: ResolvedNodeHttpHandlerConfig;
-  private readonly configProvider?: Provider<ResolvedNodeHttpHandlerConfig>;
+  private readonly configProvider: Promise<ResolvedNodeHttpHandlerConfig>;
+
   // Node http handler is hard-coded to http/1.1: https://github.com/nodejs/node/blob/ff5664b83b89c55e4ab5d5f60068fb457f1f5872/lib/_http_server.js#L286
   public readonly metadata = { handlerProtocol: "http/1.1" };
 
   constructor(options?: NodeHttpHandlerOptions | Provider<NodeHttpHandlerOptions | void>) {
-    if (typeof options === "function") {
-      this.configProvider = async () => {
-        return this.resolveDefaultConfig(await options());
-      };
-    } else {
-      this.config = this.resolveDefaultConfig(options);
-    }
+    this.configProvider = new Promise(async (resolve) => {
+      if (typeof options === "function") {
+        resolve(this.resolveDefaultConfig(await options()));
```

Comment:
Using async inside Promise constructor is an [anti-pattern](https://stackoverflow.com/a/43050114)

```suggestion
    this.configProvider = new Promise((resolve) => {
      if (typeof options === "function") {
        options().then((options) => resolve(this.resolveDefaultConfig(options)));
```